controllers/vspherecluster_controller.go

```diff
@@ -296,11 +328,39 @@ func (r clusterReconciler) reconcileAPIEndpoints(ctx *context.ClusterContext) er
 		// for this VSphereCluster into the analogous CAPI Cluster using an
 		// UnstructuredReader.
 		ctx.VSphereCluster.Status.APIEndpoints = []infrav1.APIEndpoint{apiEndpoint}
+		vsphereCluster := ctx.VSphereCluster.DeepCopy()
+
+		// Enqueue a reconcile request for the cluster when the target API
+		// server is online.
+		go func() {
+			for {
```

Comment:
You may want to look into `wait.UntilWithContext()` for something like this, if you're looking for a helper func. Certainly not required, though.